The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9653936674/job/26627182416) [(plain)](https://github.com/rust-lang/rust/commit/878a679607b61fbe42b36565964838f21d1853e1/checks/26627182416/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.421   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.434 Collecting boolean-py==4.0
#13 0.438   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.451 Collecting chardet==5.1.0
---
#13 3.461 Building wheels for collected packages: reuse
#13 3.462   Building wheel for reuse (pyproject.toml): started
#13 3.792   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.793   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.793   Stored in directory: /tmp/pip-ephem-wheel-cache-q7uvvcll/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.795 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.817   Attempting uninstall: setuptools
#13 3.817     Found existing installation: setuptools 59.6.0
#13 3.819     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.819     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.819     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.490 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.491 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.005 Collecting virtualenv
#13 5.043   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 5.111      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 86.1 MB/s eta 0:00:00
#13 5.163 Collecting platformdirs<5,>=3.9.1
#13 5.167   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.183 Collecting distlib<1,>=0.3.7
#13 5.187   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.226 Collecting filelock<4,>=3.12.2
#13 5.230   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 5.230   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 5.320 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.491 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 5.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      178112 kB
DirectMap2M:     7161856 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/d8d5732456d375f7c4bdc2f6ad771989a5e0ae02/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-d8d5732456d375f7c4bdc2f6ad771989a5e0ae02-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
fmt check
fmt: checked 5148 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/tests/ui/simd/dont-invalid-bitcast-x86_64.rs:20: line longer than 100 chars
##[error]tidy error: /checkout/tests/ui/simd/dont-invalid-bitcast-x86_64.rs:21: line longer than 100 chars
##[error]tidy error: /checkout/tests/ui/simd/dont-invalid-bitcast-x86_64.rs:24: line longer than 100 chars
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (24.1)
some tidy checks failed
Build completed unsuccessfully in 0:01:09
  local time: Mon Jun 24 23:44:23 UTC 2024
  network time: Mon, 24 Jun 2024 23:44:23 GMT
```

</details>

--- a/tests/ui/simd/dont-invalid-bitcast-masks.rs
+++ b/tests/ui/simd/dont-invalid-bitcast-masks.rs
@@ -1,0 +1,17 @@
+//@ build-pass
+//@ compile-flags: -Copt-level=3
+
+// regression test for https://github.com/rust-lang/rust/issues/110722
+// in --release we were optimizing to invalid bitcasts, due to a combination of MIR inlining and
+// mostly bad repr(simd) lowering which prevented even basic splats from working
+#![crate_type = "rlib"]
+#![feature(portable_simd)]
+use std::simd::*;
+use std::simd::num::*;
+
+pub unsafe fn mask_to_array(mask: u8) -> [i32; 8] {
+    let mut output = [0; 8];
+    let m = masksizex8::from_bitmask(mask as _);
+    output.copy_from_slice(&m.to_int().cast::<i32>().to_array());
+    output
+}

--- a/tests/ui/simd/dont-invalid-bitcast-x86_64.rs
+++ b/tests/ui/simd/dont-invalid-bitcast-x86_64.rs
@@ -1,0 +1,27 @@
+//@ build-pass
+//@ compile-flags: -Copt-level=3
+//@ only-x86_64
+// ignore-tidy-linelength
+
+// regression test for https://github.com/rust-lang/rust/issues/110707
+// in --release we were optimizing to invalid bitcasts, due to a combination of MIR inlining and
+// mostly bad repr(simd) lowering which prevented even basic splats from working
+
+#![crate_type = "rlib"]
+#![feature(portable_simd)]
+use std::simd::*;
+use std::arch::x86_64::*;
+
+#[target_feature(enable = "sse4.1")]
+pub unsafe fn fast_round_sse(i: f32x8) -> f32x8 {
+    let a = i.to_array();
+    let [low, high]: [[f32; 4]; 2] =
+        unsafe { std::mem::transmute::<[f32; 8], [[f32; 4]; 2]>(a) };
+
+    let low = f32x4::from(_mm_round_ps::<{_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC}>(f32x4::from_array(low).into()));
+    let high = f32x4::from(_mm_round_ps::<{_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC}>(f32x4::from_array(high).into()));
+
+    let a: [f32; 8] =
+        unsafe { std::mem::transmute::<[[f32; 4]; 2], [f32; 8]>([low.to_array(), high.to_array()]) };
+    f32x8::from_array(a)
+}

